datasync/chaindatafetcher/kas/repository_transactions.go

```diff
@@ -0,0 +1,178 @@
+package kas
+
+import (
+	"fmt"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"strings"
+)
+
+// TxFilteringTypes filters types which are only stored in KAS database.
+var TxFilteringTypes = map[types.TxType]bool{
+	types.TxTypeLegacyTransaction: true,
+
+	types.TxTypeValueTransfer:                      true,
+	types.TxTypeFeeDelegatedValueTransfer:          true,
+	types.TxTypeFeeDelegatedValueTransferWithRatio: true,
+
+	types.TxTypeValueTransferMemo:                      true,
+	types.TxTypeFeeDelegatedValueTransferMemo:          true,
+	types.TxTypeFeeDelegatedValueTransferMemoWithRatio: true,
+
+	types.TxTypeSmartContractDeploy:                      true,
+	types.TxTypeFeeDelegatedSmartContractDeploy:          true,
+	types.TxTypeFeeDelegatedSmartContractDeployWithRatio: true,
+
+	types.TxTypeSmartContractExecution:                      true,
+	types.TxTypeFeeDelegatedSmartContractExecution:          true,
+	types.TxTypeFeeDelegatedSmartContractExecutionWithRatio: true,
+}
+
+// transformToTxs transforms a chain event to transaction list according to the KAS database scheme.
+func transformToTxs(event blockchain.ChainEvent) []*Tx {
+	var txs []*Tx
+	// TODO-ChainDataFetcher extract EOAs in order to expire the KAS cache
+	//expiredEOAs := make(map[common.Address]struct{})
+
+	block := event.Block
+	head := block.Header()
+	receipts := event.Receipts
+
+	for idx, rawTx := range block.Transactions() {
+		txId := head.Number.Int64()*maxTransactionCount*maxTransactionLogCount + int64(idx)*maxInternalTransactionCount
```

Comment:
I don't get this equation. Where does it from?

Reply:
This is a transaction id for uniqueness.